Add a helm chart version with some backed rules

pipo02mix opened the pull request:
Start using helm package manager to maintain the resources and update components. Add some basic rules to Prometheus by default.

Towards https://github.com/giantswarm/giantswarm/issues/4534

JosephSalisbury commented on helm/README.md:
```diff
@@ -0,0 +1,217 @@
+# Prometheus
+
+Here it is a new opinionated [Prometheus](https://prometheus.io/) recipe thought to be deployed in the Giant Swarm tenant clusters.
```
```suggestion
An opinionated [Prometheus](https://prometheus.io/) Helm chart for deployment in Giant Swarm clusters.
```

JosephSalisbury commented on helm/README.md:
```diff
@@ -0,0 +1,217 @@
+
+## Prerequisites
+
+- Kubernetes 1.3+ with Beta APIs enabled
```
Beta APIs?

pipo02mix replied:
Copied from the original stable chart. I guess at that time you needed beta api gates

JosephSalisbury commented on helm/README.md:
````diff
@@ -0,0 +1,217 @@
+To uninstall/delete the deployment:
+
+```console
+$ helm delete prometheus
````
Do we want `--purge` here as well?

pipo02mix commented:
Yeah, it is huge, I can split it. Indeed it is the stable chart from the community but I made some arrangements in order to work against our rules (I added also some of our rules). I have removed the exporters because they are already in the tenant clusters running, so we don't copy configuration (ex. `kubernetes-node-exporter` repo already has this configuration)

puja108 commented on helm/README.md:
```diff
@@ -0,0 +1,217 @@
+# Prometheus
+
+An opinionated [Prometheus](https://prometheus.io/) Helm chart for deployment in Giant Swarm clusters.
```
Can we add that this is based off of upstream stable chart or so, we should give credit

pipo02mix replied:
done in the bottom

puja108 commented on helm/README.md:
````diff
@@ -0,0 +1,217 @@
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
+
+```console
+$ helm install stable/prometheus --name prometheus \
````
```suggestion
$ helm install ./helm --name prometheus \
```

puja108 commented on helm/README.md:
````diff
@@ -0,0 +1,217 @@
+Alternatively, a YAML file that specifies the values for the above parameters can be provided while installing the chart. For example,
+
+```console
+$ helm install stable/prometheus --name prometheus -f values.yaml
````
```suggestion
$ helm install ./helm --name prometheus -f values.yaml
```

puja108 commented on helm/README.md:
```diff
@@ -0,0 +1,217 @@
+to manually create a networkpolicy which allows it.
+
+
+__Note__: This chart configuration is highly influences by the [community one](https://github.com/helm/charts/tree/master/stable/prometheus).
```
```suggestion
__Note__: This chart is based off of the [upstream community chart](https://github.com/helm/charts/tree/master/stable/prometheus).
```

puja108 commented on helm/README.md:
````diff
@@ -0,0 +1,217 @@
+To install the chart:
+
+```console
+$ helm install ./helm --name prometheus
````
should we change `prometheus` to `my-prometheus` to show that you can use your own naming here?

pipo02mix replied:
good idea

pipo02mix commented on helm/README.md:
````diff
@@ -0,0 +1,217 @@
+To install the chart:
+
+```console
+$ helm install ./helm --name prometheus
````
```suggestion
$ helm install ./helm --name my-prometheus
```

pipo02mix commented on helm/README.md:
````diff
@@ -0,0 +1,217 @@
+To uninstall/delete the deployment:
+
+```console
+$ helm delete prometheus --purge
````
```suggestion
$ helm delete my-prometheus --purge
```

pipo02mix commented on helm/README.md:
````diff
@@ -0,0 +1,217 @@
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
+
+```console
+$ helm install ./helm --name prometheus \
````
```suggestion
$ helm install ./helm --name my-prometheus \
```

pipo02mix commented:
Comments addressed PTAL
